client/src/providers/AuthProvider.tsx

```diff
@@ -51,20 +56,17 @@ function AuthProvider({children, initialAuthUser}: Props): React.ReactElement {
   const [meQueryReference, loadMeQuery, disposeMeQuery] =
     useQueryLoader<AuthProviderMeQuery>(meQuery);
 
-  const {resetRelayEnvironment} = useResettableRelayContext();
-
   const signOutAsync = async (): Promise<void> => {
-    const cleanup = async (): Promise<void> => {
-      await AsyncStorage.removeItem('token');
-
-      disposeMeQuery();
+    await AsyncStorage.removeItem('token');
 
-      setUser(null);
+    disposeMeQuery();
 
-      resetRelayEnvironment();
-    };
+    commitLocalUpdate(relayEnvironment, (store) => {
+      const root = store.getRoot();
+      root.invalidateRecord();
+    });
 
```

Comment:
I implemented `RelayResettableProvider` because there was no easy way to clean all cache records.
https://stackoverflow.com/questions/45361822/how-to-invalidate-relays-cache-after-mutation
This is a post from more than 4 years ago, but it pretty much summarizes my intention for `RelayResettableProvider`.
Are you sure `root.invalidateRecord` will clean up everything?

Reply:
@0916dhkim Actually, I doesn't mean all records are cleaned up, but mark all rescords as `stale` then, `Relay` will send a request to a server to refresh the records with new one when signin again.  and It works fine in my env 😄

Reply:
@DevYuns The code looks good, so I just need to test this on my machine tonight. I will follow up after I check that the old cache is not reused.